src/Chapter1.hs

```diff
@@ -554,7 +555,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z 
+    | x > y && x < z = x
+    | x > z && x < y = x
+    | y > x && y < z = y
+    | y > z && y < x = y
+    | z > x && z < y = z
+    | z > y && z < x = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂

In this case, last two clauses can be replaced by a single:

```suggestion
    | otherwise = z
```

As at this point, this is the only possible result this function can return, so no need to make these checks 🙂